File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.cpp
class Solution {
public:
    int lenLongestFibSubseq(vector<int>& arr) {
        int n = arr.size();
        int res = 0;
        unordered_set<int> lookup(arr.begin(), arr.end()); 
        for (int i = 0 ; i < n ; i++){
            for (int j = i + 1 ; j < n ; j++){
                int curr = 2;
                int n1 = arr[i];
                int n2 = arr[j];
                while (lookup.find(n1 + n2) != lookup.end()){
                    auto temp = n1;
                    n1 = n2;
                    n2 = n2 + temp;
                    curr = curr + 1;
                }
            if (res < curr){
                res = curr;
            }
            }

        }
        return (res >= 3) ? res : 0;
    }
};

File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.py
class Solution:
    def lenLongestFibSubseq(self, arr: List[int]) -> int:
        res = 0
        lookup = set(arr)
        memo = {}
        for i in range(len(arr)):
            for j in range(i+1, len(arr)):
                curr = 2
                n1, n2 = arr[i], arr[j]

                while n1 + n2 in lookup:
                    if (n1, n2) in memo:
                        curr += memo[(n1, n2)] - 1
                        break
                    memo[(n1, n2)] = curr
                    n1, n2 = n2, n1 + n2
                    curr += 1
                res = res if res > curr else curr
        return res if res >= 3 else 0
